include/fmt/std.h

```diff
@@ -69,6 +70,33 @@ struct formatter<std::filesystem::path, Char>
         basic_string_view<Char>(quoted.data(), quoted.size()), ctx);
   }
 };
```

Comment:
I don't think this conditional branch is needed because character types other than char and wchar_t won't work anyway.

Reply:
Are you writing about a generic implementation?
Why?
``path::string<Char>()`` (and path formatter) supports ``char``, ``wchar_t``, ``char8_t``, ``char16_t``, and ``char32_t`` types.

Reply:
Ah, you are right. `string<Char>()` does the magic. In this case I suggest keeping just the generic branch and not introducing a workaround for broken msvc 2017. If anyone wants to format a path on a broken compiler version they can always use `.string()`.

Reply:
At least according to `std::format` specification only `char` and `wchar_t` are supported for `charT`, that's the reason for my suggested workaround in this case. According to my experience, MSVC 2017 is still heavily in use. If you still prefer giving up 2017 support, may I suggest then a revised solution that still detects the problematic MSVC version and does *not* provide the partial specialization in this case?

Reply:
> If you still prefer giving up 2017 support, may I suggest then a revised solution that still detects the problematic MSVC version and does not provide the partial specialization in this case?

Yes, I think it's better to disable `path` support on an older broken compiler completely rather than enable it partially for some character types. Detecting the problematic MSVC version sounds reasonable but maybe move it to a separate PR and leave this one to fixing 2019+?

Reply:
Since I now need an extra conditional for the unit test anyway to *exclude* testing this scenario for Visual Studio 2017, which value would it have to keep the partial specialization without conditional?

Reply:
Why disable path support if ranges are broken in MSVC 2017?

If I understand correctly, then the problem will be with formatter for any type that has member functions ``begin()`` and ``end()``. Or not?

Reply:
I'm not suggesting to disable path support unconditionally. I'm suggesting to *enable* it conditionally for all cases that we know, so effectively *not* providing any specialization if `FMT_MSC_VERSION && FMT_MSC_VERSION < 1920`. 
The same conditional will need to be added to the ranges-std test anyway, so why providing functionality that we decide to not test, because it does not work? Does that make sense to you?

Reply:
> which value would it have to keep the partial specialization without conditional?

It will keep the implementation simple if I understood your question correctly. We generally try to avoid conditional compilation unless absolutely necessary and limiting it to test significantly reduces the scope.

Reply:
> it's better to disable path support on an older broken compiler completely rather than enable it partially for some character types

My revised solution wasn't suggesting to enable it partially for some character types.

Reply:
The revised solution seems OK to me.

Reply:
Done